lib/stdlib/src/io_lib_format.erl

```diff
@@ -557,250 +557,251 @@ float_data([_|Cs], Ds) ->
-
-vmIsTrailingZero(Vm, Vr, _Vp, false = _VmTZ, VrTZ, Removed, LastRemovedDigit) ->
-    handle_50_dotdot_0(Vm, Vr, VrTZ, Removed, LastRemovedDigit);
-vmIsTrailingZero(Vm, Vr, _Vp, _VmTZ, VrTZ, Removed, LastRemovedDigit) when (Vm rem 10) /= 0 ->
-    handle_50_dotdot_0(Vm, Vr, VrTZ, Removed, LastRemovedDigit);
-vmIsTrailingZero(Vm, Vr, Vp, VmTZ, VrTZ, Removed, LastRemovedDigit) ->
-    vmIsTrailingZero(Vm div 10, Vr div 10, Vp div 10, VmTZ, LastRemovedDigit == 0 andalso VrTZ, 1 + Removed, Vr rem 10).
-
-handle_50_dotdot_0(Vm, Vr, true, Removed, 5) when (Vr rem 2) =:= 0 ->
-    {Vm, Vr, Removed, 4};
-handle_50_dotdot_0(Vm, Vr, _VrTZ, Removed, LastRemovedDigit) ->
-    {Vm, Vr, Removed, LastRemovedDigit}.
-
-handle_zero_output_mod(_Vr, _Vm, _Accept, _VmTZ, LastRemovedDigit) when LastRemovedDigit >= 5 ->
-    1;
-handle_zero_output_mod(Vr, Vm, Accept, VmTZ, _LastRemovedDigit) when Vr =:= Vm, ((not Accept) or (not VmTZ)) ->
-    1;
-handle_zero_output_mod(_Vr, _Vm, _Accept, _VmTZ, _LastRemovedDigit) ->
-    0.
-
-insert_decimal(Place, S, Float) ->
-    L = length(S),
-    Exp = Place + L - 1,
-    ExpL = integer_to_list(Exp),
-    ExpCost = length(ExpL) + 2,
-    if
-        Place < 0 ->
-            if
-                Exp >= 0 ->
-                    {S0, S1} = lists:split(L + Place, S),
-                    S0 ++ "." ++ S1;
-                2 - Place - L =< ExpCost ->
-                    "0." ++ lists:duplicate(-Place - L, $0) ++ S;
-                true ->
-                    insert_exp(ExpL, S)
-            end;
-        true ->
-            Dot = if L =:= 1 -> 1; true -> 0 end,
-            if
-                %% All integers in the range [-2^53, 2^53] can
-                %% be stored without loss of precision in an
-                %% IEEE 754 64-bit double but 2^53+1 cannot be
-                %% stored in an IEEE 754 64-bit double without
-                %% loss of precision (float((1 bsl 53)+1) =:=
-                %% float(1 bsl 53)). It thus makes sense to
-                %% show floats that are >= 2^53 or <= -2^53 in
-                %% scientific notation to indicate that the
-                %% number is so large that there could be loss
-                %% in precion when adding or subtracting 1.
-                %%
-                %% https://stackoverflow.com/questions/1848700/biggest-integer-that-can-be-stored-in-a-double?answertab=votes#tab-top
-                ExpCost + Dot >= Place + 2 andalso abs(Float) < float(1 bsl 53) ->
-                    S ++ lists:duplicate(Place, $0) ++ ".0";
-                true ->
-                    insert_exp(ExpL, S)
-            end
-    end.
-
-
-insert_exp(ExpL, [C]) ->
-    [C] ++ ".0e" ++ ExpL;
-insert_exp(ExpL, [C | S]) ->
-    [C] ++ "." ++ S ++ "e" ++ ExpL.
-
-insert_minus(0, Digits) ->
-    Digits;
-insert_minus(1, Digits) ->
-    [$-] ++ Digits.
+    float_to_list(Float, [pretty_print]).
+    % case sign_mantissa_exponent(Float) of
```

Comment:
I don't think there is much value in keeping the Erlang implementation as comments. If it is kept here, we should describe in a comment why it is kept. Perhaps it could be useful to add the Erlang implementation to the test suite so that it can be used to test the C implementation but I'm not sure as I think we have other ways of testing that the generated string is correct, right?

Reply:
We do for fwrite_g/1. I kept it around mostly as a way to to test stuff and benchmark so far. Was waiting for the last minute

Open question while i am at it: what about using float_to_binary here ? as in theory io:format return io_data iirc and that would be a _massive_ memory change. We have problem in some library like Jason where encoding a medium size payload with some floats OOM-kill the BEAM due to that...

Reply:
Good question. I think you mean `io_lib:format` (`io:format` returns ok). `io_lib:format` returns a list of characters so I don't think that would work. But Jason can start to call `float_to_binary(F, [short])` instead of `fwrite_g` as soon as it is available, right?

Reply:
Ah yes ok i think i did a mix between two things.

The problem of using the BIF is that the only way around will be a check on OTP version, but fair indeed.